### PR TITLE
[spark] implement of `rollback` action using Spark Procedure

### PR DESCRIPTION
**Pull request by lvyanquan**
### Purpose

implement for this https://github.com/apache/incubator-paimon/issues/1896 Feature request after this https://github.com/apache/incubator-paimon/pull/1785 pr.
As it may be the first implementation, Hope to get some guidance😊.
Will try to add more implementations later.

@JingsongLi  @SteNicholas looking forward to your opinion.

### Tests
add some new test classes like 
[RollbackProcedureTest.scala](https://github.com/apache/incubator-paimon/pull/1897/files#diff-996f2c6f25319cd2ff34984e540b749449b8250ebcbc05ff9b482c672d83d8b6)
[CreateAndDeleteTagProcedureTest.scala](https://github.com/apache/incubator-paimon/pull/1897/files#diff-8ef36496ed0a33b9f1944112bba5301387fabd94c361d67877a1d42bf1961c6a)
include test cases for both success and failure.

### API and Format

Add new spark procedures: rollback to tag/snapshot、create tag、 delete tag.
No user interface change.

### Documentation
under doc folder

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/BaseProcedure.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * get table by database and table name.
+     *
+     * @param database if null, use current database
+     * @param table -
+     * @return -
+     */
+    public SparkTable getTable(String database, String table) {
+        Preconditions.checkArgument(table != null && !table.trim().isEmpty(), "tableName is empty");
+        CatalogManager catalogManager = spark.sessionState().catalogManager();
+        SparkCatalog sparkCatalog = new SparkCatalog();
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/BaseProcedure.java:46]: Create catalog per call is not good.
For example, hive connector will be leaked.

`ProcedureBuilder` already have `TableCatalog`.

Comment (lvyanquan) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/BaseProcedure.java]: address it.

**Review by JingsongLi — CHANGES_REQUESTED** [review]
You can delete useless codes.

**Comment by lvyanquan** [comment]
rebase branch to avoid CI failure and address comment.

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkUtils.java [hunk 2]
```diff
@@ -29,4 +44,112 @@ public static IOManager createIOManager() {
...
+     * A modified version of Spark's LookupCatalog.CatalogAndIdentifier.unapply Attempts to find the
+     * catalog and identifier a multipart identifier represents.
+     *
+     * @param nameParts Multipart identifier representing a table
+     * @return The CatalogPlugin and Identifier for the table
+     */
+    public static <C, T> Pair<C, T> catalogAndIdentifier(
+            List<String> nameParts,
+            Function<String, C> catalogProvider,
+            BiFunction<String[], String, T> identiferProvider,
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkUtils.java:77]: identifierProvider

**Comment by JingsongLi** [comment]
Just left one minor comment.

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkUtils.java (hunk 2 above)
Comment (lvyanquan) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkUtils.java]: done.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**